example/bwmf/bwmf_task.go

```diff
@@ -154,8 +282,12 @@ func (t *bwmfTask) DataReady(ctx context.Context, fromID uint64, method string,
 }
 func (t *bwmfTask) doDataReady(ctx context.Context, fromID uint64, method string, output proto.Message) {
 	t.logger.Printf("doDataReady, task %d, from %d, epoch %d, method %s", t.taskID, fromID, t.epoch, method)
-	t.peerShardReady[fromID] = true
-	if len(t.peerShardReady) == int(t.numOfTasks) {
+	resp, bOk := output.(*pb.Response)
+	if !bOk {
+		t.logger.Panicf("doDataRead, corruption in proto.Message.")
+	}
+	t.peerShards[resp.BlockId] = resp.Shard
+	if len(t.peerShards) == int(t.numOfTasks) {
 		if t.epoch%2 == 0 {
 			t.logger.Printf("Full D ready, task %d, epoch %d", t.taskID, t.epoch)
 			go t.updateTShard(ctx)
```

Comment:
@BaiGang ,

can you remove the "go" statement before updateD/TShard. It can simplify our case here. We can assume no failure case now.